site/src/main/java/uk/nhs/digital/ps/beans/Dataset.java

```diff
@@ -55,23 +55,23 @@ public String getGeographicCoverage() {
     }
 
     /*
-     * Migration: 20171208
-     * New field: publicationsystem:Files-v2
-     * Old field: publicationsystem:Files
+     * Migration: 20171213
+     * New field: publicationsystem:Files-v3
+     * Old field: publicationsystem:Files-v2
      *
-     * Temporary make this getter forward and backward compatible, meaning we can deploy this code without running
-     * content migration.
+     * Temporary supporting old and new versions until all the live content is migrated to the new way
      */
+    @HippoEssentialsGenerated(internalName = PropertyKeys.FILES_V3)
+    public List<Attachment> getFiles() {
+        assertPropertyPermitted(PropertyKeys.FILES_V3);
+
+        return getChildBeansByName(PropertyKeys.FILES_V3, Attachment.class);
+    }
     @HippoEssentialsGenerated(internalName = PropertyKeys.FILES_V2)
-    public List<HippoResourceBean> getFiles() {
+    public List<HippoResourceBean> getFilesOld() {
```

Comment:
Reference to this method needs to be added to the JIRA ticket referencing all bits of code that need removing once the data upgrade scripts are run (see comment in the review of `Publication.java`).

Reply:
Created RPS-238